### DISCUSSION TITLE
web3Auth social login without popup & RainbowKit as external wallet development issue

### DISCUSSION THREAD

**Discussion by Le4kno3** [Web3Auth/Web3Auth#1340] [Help]
Hi Web3Auth Team,

I am trying to implement a login feature similar to https://app.decentral.games/

Details
- It should not show popup directly when users click login, it should show similar to what decentral (https://app.decentral.games/) implemented. It should show the 4-5 major social media as a "continue with" butttons. Instead of a presenting the same in a popup
- I also want users to connect their existing wallet using RainbowKit. Yes this is a different product but in docs I saw some integration so I though to mention this.
- If possible, then both of these logins (web3auth and rainbowkit) is wrapped in wagmi. Else if it cannot be done then atleast the web3auth part of non-popup should be wrapped in wagmi.

Solution as per my understanding
1) Both web3auth and rainbowkit are different solutions, so we cannot bring in the same wagmi wrapper.
2) web3auth object has more features than wagmi react hooks, so I am thinking of sharing this web3auth object to all using Context API or Recoil library.
3) For RainbowKit, I can wrap it in wagmi, as almost all query can be done by wagmi react hooks for already created wallets.

Pressing Requirement
- I want to have a non-popup social media web3auth logins buttons without using the "custom authentication", where I need to set each social media by clientID and Oauth key differently. Instead I want web3auth to do so like by using web3auth/modal.[

the sign in button, before click
![photo_2023-02-17_21-56-08](https://user-images.githubusercontent.com/17622439/219863818-5f0bf111-e28d-4f5a-9ede-f9f0d771de0f.jpg)

the interface "non-popup" after click.
![photo_2023-02-17_21-56-27](https://user-images.githubusercontent.com/17622439/219863821-7822c070-1af1-4a03-8bec-56c1a48373d7.jpg)
](url)

**Comment by MeeTeeT** [comment]
Hi, dis you find a way to do it ? 
Facing the same question...

THanks

**Reply by mayurbagga** [reply]
me tooo
